### PR TITLE
Uf 8361 add scheduled job

### PR DESCRIPTION
**Pull request by MickeWallner**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -4,7 +4,7 @@
 	<parent>
 		<groupId>se.sundsvall.dept44</groupId>
 		<artifactId>dept44-service-parent</artifactId>
-		<version>5.0.2</version>
+		<version>5.0.1</version>
```
Comment (josc04) [inline pom.xml:7]: Any reason to not use latest version? If so, maybe add comment on why?

File: src/main/java/se/sundsvall/smloader/service/scheduler/CaseProcessingScheduler.java [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+import se.sundsvall.dept44.requestid.RequestId;
+import se.sundsvall.dept44.util.jacoco.ExcludeFromJacocoGeneratedCoverageReport;
+import se.sundsvall.smloader.service.OpenEService;
+import se.sundsvall.smloader.service.SupportManagementService;
+
+import java.time.LocalDateTime;
+
+@Service
+@ConfigurationProperties("scheduler.caseprocessing.cron")
+@ExcludeFromJacocoGeneratedCoverageReport
```
Comment (josc04) [inline src/main/java/se/sundsvall/smloader/service/scheduler/CaseProcessingScheduler.java:18]: Remove exclusion?

File: src/main/java/se/sundsvall/smloader/service/scheduler/CaseProcessingScheduler.java [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+import org.springframework.stereotype.Service;
+import se.sundsvall.dept44.requestid.RequestId;
+import se.sundsvall.dept44.util.jacoco.ExcludeFromJacocoGeneratedCoverageReport;
+import se.sundsvall.smloader.service.OpenEService;
+import se.sundsvall.smloader.service.SupportManagementService;
+
+import java.time.LocalDateTime;
+
+@Service
+@ConfigurationProperties("scheduler.caseprocessing.cron")
```
Comment (josc04) [inline src/main/java/se/sundsvall/smloader/service/scheduler/CaseProcessingScheduler.java:17]: Is this needed?

File: pom.xml [hunk 4]
```diff
@@ -43,6 +45,12 @@
 			<artifactId>flyway-mysql</artifactId>
 		</dependency>
 		<!--- Other -->
+		<dependency>
```
Comment (josc04) [inline pom.xml:48]: Is this used or can it be removed?

File: src/main/resources/db/migration/V1_1__createTables.sql [hunk 5]
```diff
@@ -25,7 +25,15 @@
         primary key (family_id)
     ) engine=InnoDB;
 
+    create table shedlock(
```
Comment (josc04) [inline src/main/resources/db/migration/V1_1__createTables.sql:28]: I do not know the history of this service, but shouldn't the add of this table be in an own migration file (V1_2)? Or will you wipe the test-db and re-create all tables again?

Comment (MickeWallner) [reply src/main/resources/db/migration/V1_1__createTables.sql:28]: I'll will wipe the test-db and recreate it. The service is never deployed to prod.

File: src/main/java/se/sundsvall/smloader/service/scheduler/CaseProcessingScheduler.java [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+		this.supportManagementService = supportManagementService;
+	}
+
+	@Scheduled(cron = "${scheduler.caseprocessing.cron.expression}")
+	@SchedulerLock(name = "processcases", lockAtMostFor = "${scheduler.shedlock-lock-at-most-for}")
+	public void execute() {
+		RequestId.init();
+
+		LOGGER.info(LOG_IMPORT_STARTED);
+		openEService.fetchAndSaveNewOpenECases(LocalDateTime.now().minusDays(1).withHour(0).withMinute(0).withSecond(0).withNano(0), LocalDateTime.now());
```
Comment (ronnberg) [inline src/main/java/se/sundsvall/smloader/service/scheduler/CaseProcessingScheduler.java:37]: Fetches one day back, each day. What happens if job fails one day (service broken or something). Will those cases created that day never be fetched?

Comment (MickeWallner) [reply src/main/java/se/sundsvall/smloader/service/scheduler/CaseProcessingScheduler.java:37]: This handling is not ready yet and this is for testing purpose only. Will be managed by config when it's ready.

**Review by ronnberg — APPROVED** [review]

File: src/main/java/se/sundsvall/smloader/service/scheduler/CaseProcessingScheduler.java (hunk 6 above)
Comment (MickeWallner) [reply src/main/java/se/sundsvall/smloader/service/scheduler/CaseProcessingScheduler.java]: Added a config-attribute for days to fetch now.

File: src/main/java/se/sundsvall/smloader/service/scheduler/CaseProcessingScheduler.java [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+@Service
+public class CaseProcessingScheduler {
+
+	private static final Logger LOGGER = LoggerFactory.getLogger(CaseProcessingScheduler.class);
+	private static final String LOG_IMPORT_STARTED = "Beginning of importing cases";
+	private static final String LOG_IMPORT_ENDED = "Import of cases has ended";
+	private static final String LOG_EXPORT_STARTED = "Beginning of exporting cases";
+	private static final String LOG_EXPORT_ENDED = "Export of cases has ended";
+
+	@Value("${config.scheduler.fetch-days}")
```
Comment (ronnberg) [inline src/main/java/se/sundsvall/smloader/service/scheduler/CaseProcessingScheduler.java:24]: Default to 1?

**Review by ronnberg — APPROVED** [review]

**Pull request merged**